### ISSUE TITLE
Adding my TE translations 

### ISSUE THREAD

**Issue by neonerz** [gentlegiantJGC/Minecraft-World-Converter#3]
I wanted to go over how I'm modifying Tile Entities to get them to work on Bedrock. From there you could let me know how you'd want me to implement it. I will ignore ID changes since you seem to already have that figured out

**Banner**
- Patterns data changes from a list of strings to a list of compounds (`color:int`, `pattern:string`). I haven't looked too much into the Java data yet, but I suspect the list of string is probably a json of color and pattern.
- Invalid patterns data (i.e., java data) will cause the banner to not show up

**Skull**
- Has two new components added for the dragon skull `MouthMoving:byte` and `MouthTickCount:int`
- Seems to HAVE to have those components added or the block is invisible

**MobSpawner**
- `SpawnData` and `SpawnPotential` from Java need to be removed
- You can only define the entity type using the new component `EntityID:int`
- Spawn position, animation position, max spawn, etc are all moved to the root in new components (out of the scope to list them all)

**Things with inventories (chests, furnace, hopper, etc) "Items"**
- This seems to go for every TE with a Items component
- There is a LOT that needs to be done here due to various Bedrock differences. I'll briefly explain
- All item names need to be converted into a int using the item ID (and in some cases new DV like water/lava/milk bucket)
- Enchantments need to be remapped for new bedrock enchant IDs (including change from string to int)
- All potion IDs (drink, splash, lingering) need to be remapped for new bedrock potion IDs (some don't exist, including change from string to int)
- All tipped arrow IDs need to be remapped to new bedrock tipped arrow IDs (some don't exist, including change from string to int)
- `CanPlaceOn` and `CanDestroy` need to be moved out of tag and put into the root

**CommandBlock**
- I found some older maps from older versions of minecraft were missing the following components and causing the CommandBlock entity to be invalid. Need to make sure they exist `auto, powered, SuccessCount, conditionMe, TrackOutput`

**Sign**
- I need new handling for this. It changed from 1.1 to 1.2. Now it seems there's only a single text line delimited with what I think are line breaks (`\n`) instead of the `Text1`-`Text4` strings.

**BrewingStand**
- I'll have to look into this more. The handling I have was for 1.1 brewing stands before the current Java version was implemented in 1.2. It's possible that these might convert as is (with the ID change), but it's also possible the component names will be slightly different (they were before)

**StructureBlock**
- Probably best just to remove them for now. They are totally different and don't function in any way similar (yet I guess)

For the re-mappings needed for **Items**, I already have them mapped in a json file. I'll clean it up a bit specific for this purpose.

**Comment by gentlegiantJGC** [comment]
Right. First things first I was actually working on the item mappings yesterday. Got PC to PE mostly done.
I will do a quick mockup of how I was thinking of doing the overall structure of the code but will do a quick TLDR here. The convert function will have a section checking for every te/e ID. In each section will be code changing the things unique to that entity. There will then be other functions within the same file that deal with things like inventories

**Comment by neonerz** [comment]
So should I just start working on a bunch of functions to change the individual TE data? I'll assume we'll use your item mappings (mine is incomplete anyway), but I'll use my potion/tipped arrow/enchant mappings (since those are complete)?

Entity data is going to be a HUGE PITA, but I guess once we finish with TE, we could start working on them (though, I think it's a lot lower of a priority since entities work so differently in Bedrock, it's probably best to delete them and start from scratch anyway).
